src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -510,16 +559,22 @@ async def resolve_parsed_dependencies(
     )
 
     if parsed_imports:
+        source_address = request.field_set.address
+        # this is probably the right place to disambiguate w/ dep rules
         owners_per_import = await MultiGet(
             Get(
                 PythonModuleOwners,
                 PythonModuleOwnersRequest(imported_module, resolve=request.resolve),
             )
             for imported_module in parsed_imports
         )
+        filtered_owners_per_import = await _filter_owners_per_import(
+            field_set=request.field_set,
+            owners_per_import=owners_per_import,
+        )
         resolve_results = _get_imports_info(
             address=request.field_set.address,
-            owners_per_import=owners_per_import,
+            owners_per_import=filtered_owners_per_import,
```

Comment:
I thought about putting all of this under `_get_imports_info`, but that changes the semantics of that function, as it is now filtering and getting import info. So, that seems to break the single-responsibility-principle. So, I made `_filter_owners_per_import` a separate function.

Reply:
I agree that a separate function for `_filter_owners_per_import` makes sense. Thinking about SRP, I think that the disambiguation based on explicit dependencies might actually make more sense outside of `_get_imports_info`. see [here](https://github.com/pantsbuild/pants/pull/17631#discussion_r1038900000) about using `ImportOwnerStatus.ambiguous`